File: ServerSpace/src/main/java/Business/Tarifa.java
package Business;

import EntityLayer.EntidadEntity;
import EntityLayer.TarifaEntity;
import java.util.ArrayList;

import DataLayer.GeneralDB;
import DataLayer.TarifaDB;
import Framework.Inj;

public class Tarifa {

    public ArrayList<TarifaEntity> ObtenerMain() {
        try {
            TarifaDB BD = new TarifaDB();
            return BD.ObtenerMain();
        } catch (Exception ex) {
            throw new UnsupportedOperationException("MENSAJE :" + ex);
        }
    }

    public TarifaEntity Registrar(TarifaEntity Item) {
        try {
            Inj.IniciarTranssaccion(true);
            TarifaDB BD = new TarifaDB();
            BD.Registrar(Item);
            Inj.FinalizarTranssaccion();
            return Item;
        } catch (Exception ex) {
            throw new UnsupportedOperationException("MENSAJE :" + ex);
        }
    }

    public ArrayList<TarifaEntity> ObtenerItem(int Id) {
        try {
            TarifaDB BD = new TarifaDB();
            return BD.ObtenerItem(Id);
        } catch (Exception ex) {
            throw new UnsupportedOperationException("MENSAJE :" + ex);
        }
    }

    public ArrayList<TarifaEntity> BuscarRecurso(String Nombre, int Tipo) {
        try {
            TarifaDB BD = new TarifaDB();
            return BD.BuscarRecurso(Nombre, Tipo);
        } catch (Exception ex) {
            throw new UnsupportedOperationException("MENSAJE :" + ex);
        }
    }
     public ArrayList<TarifaEntity> GetTarifaBuscarItem(String NomComercial) {
        try {
            TarifaDB BD = new TarifaDB();
            return BD.GetTarifaBuscarItem(NomComercial);
        } catch (Exception ex) {
            System.out.println("ERROR " + ex);
            throw new UnsupportedOperationException("MENSAJE :" + ex);
        }
    }
}

File: SharedApi/Space/Space.BusinessLayer/Tarifa.cs
﻿using Framework;
using Space.DataLayer;
using Space.EntityLayer;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Space.BusinessLayer
{
    public class Tarifa
    {

        public List<TarifaEntity> ObtenerMain()
        {
            TarifaDB DB = new TarifaDB();
            return DB.ObtenerMain();
        }

        public Int32 Registrar(TarifaEntity Ent)
        {
            TarifaDB DB = new TarifaDB();
            DB.Registrar(Ent);
            return Ent.TarifaId;
        }

       

        public List<TarifaEntity> ObtenerItem(Int32 TarifaId)
        {
            TarifaDB DB = new TarifaDB();

            return DB.ObtenerItem(TarifaId);
        }

        public List<TarifaEntity> BuscarRecurso(String Nombre, Int32 Tipo)
        {
            TarifaDB DB = new TarifaDB();
            return DB.BuscarRecurso(Nombre, Tipo);
        }
    public List<TarifaEntity> GetTarifaBuscarItem(String Nombre)
        {
            TarifaDB DB = new TarifaDB();
            return DB.GetTarifaBuscarItem(Nombre);
        }

    }
}
